sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/udf/GetCosmosItemIdentityValue.scala

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.spark.udf
+
+import com.azure.cosmos.models.PartitionKey
+import com.azure.cosmos.spark.CosmosItemIdentityHelper
+import com.azure.cosmos.spark.CosmosPredicates.{requireNotNull, requireNotNullOrEmpty}
+import org.apache.spark.sql.api.java.UDF2
+
+@SerialVersionUID(1L)
+class GetCosmosItemIdentityValue extends UDF2[String, Object, String]{
+    override def call
+    (
+        id: String,
+        partitionKeyValue: Object
```

Comment:
HPK worth thinking through to see whether we need/should special case public API here for it already (even if for now we won't support it yet)

Reply:
yea, not sure which way is better, followed the feedRange UDF for now. But will reach out to discuss more

Reply:
FeedRange UDF we did not consider HPK for initially
I think I would just change the second parameter to allow always an array of pkvalues
That way we won't need two UDFs

Reply:
object can be an array as well. The code for handling HPK is 
```
    partitionKeyValue match {
      // for subpartitions case
      case wrappedArray: mutable.WrappedArray[Any] =>
        s"id($id).pk(${objectMapper.writeValueAsString(wrappedArray.toList.asJava)})"
```

and when they use the udf, they can call like following: 
`.withColumn("_itemIdentity", expr("GetCosmosItemIdentityValue(id, array(tenantId, userId, sessionId))"))`

Reply:
Makes sense